### ISSUE TITLE
why3 package cannot be pinned

### ISSUE THREAD

**Issue by jensgerlach** [ocaml/opam-repository#12369]
Hello,

trying to install version 0.88.3 of `why3` with ```opam pin``` fails immediately as shown below.
(The package is immediately unpinned again.)

```
$ opam pin add why3 0.88.3
why3 is now version-pinned to 0.88.3

why3 is now unpinned from version 0.88.3
```
Any idea why this is the case?

(I am using `opam 1.2.2` with `ocaml 4.05.0` on `xubuntu 18-04`.

**Comment by jensgerlach** [comment]
Pinning `coqide` to `8.7.2` works without problems.

**Comment by mseri** [comment]
I have seen this happening when some dependency error was happening with the pinned package. What happens if you run `opam install why3.0.88.3`?

**Comment by jensgerlach** [comment]
This installs `why3.0.88.3` without problems!

**Comment by mseri** [comment]
Then I am out of ideas for the moment, sorry

**Comment by jensgerlach** [comment]
At least I can now use why3-0.88.3

**Comment by rjbou** [comment]
The pinning problem is from why3 packaging. On debug mode, we can see that
```
error: File /opamroot/repo/default/packages/why3/why3.0.88.3/url does not exist
```
wich is confirmed in the [opam repo](https://github.com/ocaml/opam-repository/tree/master/packages/why3/why3.0.88.3) (no url file since for several packages) . As the error is raised, the pin fails and opam unpin directly the package.

**Comment by AltGr** [comment]
`version`-pins should obviously not require an url, but since other pinning kinds might, that could explain the bug. This is long fixed in opam 2.0 branch, in any case!
Thanks for reporting.
